roles/openshift_loadbalancer/tasks/main.yml

```diff
@@ -1,14 +1,44 @@
 ---
-- fail: msg="Cannot use containerized=true for load balancer hosts."
-  when: openshift.common.is_containerized | bool
-
 - name: Install haproxy
   package: name=haproxy state=present
+  when: not openshift.common.is_containerized | bool
+
+- name: Pull haproxy image
+  command: >
+    docker pull {{ openshift.common.router_image }}:{{ openshift_image_tag }}
+  when: openshift.common.is_containerized | bool
+
+- name: Create config directory for haproxy
+  file:
+    path: /etc/haproxy
+    state: directory
+  when: openshift.common.is_containerized | bool
+
+- name: Create the systemd unit files
+  template:
+    src: "haproxy.docker.service.j2"
+    dest: "{{ containerized_svc_dir }}/openshift_loadbalancer.service"
+  when: openshift.common.is_containerized | bool
+  notify: restart haproxy
+
+- name: Configure systemd service directory for haproxy
+  file:
+    path: /etc/systemd/system/haproxy.service.d
+    state: directory
+  when: not openshift.common.is_containerized | bool
+
+- name: Create the systemd unit files
```

Comment:
this was from the original file https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_loadbalancer/tasks/main.yml